src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java

```diff
@@ -308,7 +303,7 @@ private void closeAndThrow(final Model model, final String message) throws Kettl
     }
 
     private Object convertSqlValueToRdf(final Object sqlValue, @Nullable final RDFDatatype rdfDatatype) {
-        if (rdfDatatype == null || rdfDatatype.getURI().equals(XSD_NAMESPACE_IRI + "#string")) {
+        if (rdfDatatype == null || rdfDatatype.equals(XSDDatatype.XSDstring)) {
```

Comment:
@langsamu It's a nice idea. But... have you tested this?

I am concerned that `RDFDatatype` does not override the object equality method `equals` so you end up with the default from `Object#equals(Object)` which is instead instance equality i.e. `a == b`.

As long as there is only a single instance of each RDFDatatype in the JVM then that will be fine... but I am not certain that is the case.

Reply:
That `rdfDatatype` comes from `addResourceProperties` via [`TypeMapper`](https://jena.apache.org/documentation/javadoc/jena/org/apache/jena/datatypes/TypeMapper.html), "a global registry of known datatypes" based on a (URI) string hash:
https://github.com/adamretter/kettle-jena-plugins/blob/fee4214032c2d1f1b8b5b052f002e08a312e6685/src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java#L286

TypeMapper loads XSD simple types
https://github.com/apache/jena/blob/9fb9f050090c8fabbcca9552ab98a517aae78449/jena-core/src/main/java/org/apache/jena/datatypes/TypeMapper.java#L72

using the same `XSDDatatype` we're comparing to
https://github.com/apache/jena/blob/9fb9f050090c8fabbcca9552ab98a517aae78449/jena-core/src/main/java/org/apache/jena/datatypes/xsd/XSDDatatype.java#L589

so yes, they're the same.

Reply:
Okay sounds good.